### PR TITLE
stunnel 5.67 support

### PR DESCRIPTION
**Pull request by julek-wolfssl**
# Description

- Add support for external data callbacks for WOLFSSL_SESSION
- ssl->session->ticketAdd may be set when parsing a ticket before PSK
- Add option to prioritise PSK order instead of ciphersuite order
- Update ctx->method->version when using Set_CTX_max_proto_version API
- Simplify wolfSSL_parse_cipher_list
  - Keep copy of old list and then add in the previous ciphersuites depending on whether we are doing only TLS 1.3 ciphersuites or not
- Specify CRL revocation in alert
  - Update reason string to match OpenSSL

Fixes zd#15433

# Testing

Added internal tests for all changes. Passes stunnel tests.

# Checklist

 - [x] added tests
 - [ ] updated/added doxygen
 - [ ] updated appropriate READMEs
 - [ ] Updated manual and documentation

### CODE REVIEW

**Comment by kaleb-himes** [comment]
Jenkins is unable to test in current state for the most part, it is reporting merge conflicts. Can you rebase with master when you get a chance?

**Comment by julek-wolfssl** [comment]
I'm pretty sure this is rebased on latest master. I haven't found which Jenkins case you're talking about. I was expecting some failures due to the amount of our configs. I'll fix them and then see about the merge conflicts you're seeing.

**Review by dgarske — APPROVED** [review]
The work looks great! Please rebase, squash merge and force push. Once CI is happy I will merge.

**Comment by kaleb-himes** [comment]
retest this please

**Comment by julek-wolfssl** [comment]
@dgarske Squashed, rebased, and CI is happy.

File: src/ssl.c [hunk 1]
```diff
@@ -12565,10 +12457,13 @@ static int CheckcipherList(const char* list)
...
         const char* list)
 {
-    int       ret          = 0;
-    int listattribute = 0;
-    char*     buildcipherList = NULL;
-    int tls13Only = 0;
+    int     ret = 0;
+    int     listattribute = 0;
+    int     tls13Only = 0;
+    byte    suitesCpy[WOLFSSL_MAX_SUITE_SZ];
```
Comment (dgarske) [inline src/ssl.c:12463]: The `WOLFSSL_MAX_SUITE_SZ` is 300 bytes and will blow out small stack. Please add small stack support for it.

File: src/ssl.c [hunk 2]
```diff
@@ -31630,6 +31652,78 @@ int wolfSSL_CTX_use_PrivateKey(WOLFSSL_CTX *ctx, WOLFSSL_EVP_PKEY *pkey)
     defined(WOLFSSL_HAPROXY) || defined(OPENSSL_EXTRA) || \
     defined(HAVE_LIGHTY)) || defined(HAVE_EX_DATA) || \
     defined(WOLFSSL_WPAS_SMALL)
+CRYPTO_EX_cb_ctx* crypto_ex_cb_ctx_session = NULL;
+
+static int crypto_ex_cb_new(CRYPTO_EX_cb_ctx** dst, long ctx_l, void* ctx_ptr,
```
Comment (dgarske) [inline src/ssl.c:31657]: Should these new `crypto_ex` functions have a leading `wolfssl_`?

**Review by dgarske — CHANGES_REQUESTED** [review]
Lots of great work here! Just a couple minor things.

File: src/ssl.c (hunk 2 above)
Comment (julek-wolfssl) [reply src/ssl.c]: They are static and internal functions so I don't think that is necessary.

File: src/ssl.c (hunk 1 above)
Comment (julek-wolfssl) [reply src/ssl.c:12463]: Added small stack option.

**Review by dgarske — APPROVED** [review]

**Pull request merged**